includes/class-algolia-utils.php

```diff
@@ -206,21 +209,26 @@ public static function explode_content( $content ) {
             $max_size = (int) ALGOLIA_CONTENT_MAX_SIZE;
         }
 
-	    $parts = array();
+        $parts = array();
+        $prefix = '';
         while ( true ) {
             $content = trim( (string) $content );
-            if ( mb_strlen( $content ) <= $max_size ) {
-                $parts[] = $content;
+            if ( strlen( $content ) <= $max_size ) {
+                $parts[] = $prefix . $content;
 
                 break;
             }
 
-            $cutAtPosition = mb_strpos( $content, ' ', $max_size );
+            $offset = -( strlen( $content ) - $max_size );
+            $cutAtPosition = strrpos( $content, ' ', $offset);
```

Comment:
Here is where the magic happens @julienbourdeau .
We detect the last occurrence of the space char before the max size is reached.